crates/cdk/src/wallet/payment_request.rs

```diff
@@ -0,0 +1,160 @@
+//! Utilities for paying NUT-18 Payment Requests.
+//!
+//! This module prepares and broadcasts payments for Cashu NUT-18 payment requests using either
+//! Nostr or HTTP transports when available. If no transport is present in the request, an error
+//! is returned so callers can handle alternative delivery mechanisms explicitly.
+
+use cdk_common::{Amount, PaymentRequest, PaymentRequestPayload, TransportType};
+use nostr_sdk::nips::nip19::Nip19Profile;
+use nostr_sdk::{Client as NostrClient, EventBuilder, FromBech32, Keys};
+use reqwest::Client;
+
+use crate::wallet::SendOptions;
+use crate::Wallet;
+
+/// Pay a NUT-18 PaymentRequest using a specific wallet.
+///
+/// - If the request contains a Nostr or HttpPost transport, it will try those (preferring Nostr).
+/// - If no usable transport is present, this returns an error.
+/// - If the request has no amount, a `custom_amount` must be provided.
+pub async fn pay_request(
```

Comment:
I think we should impl this on the wallet and have it take &self like most of out other fns.

so the caller is `self.pay_request(payment_request, amount).await?`